src/arrays.js

```diff
@@ -8,33 +8,69 @@ const each = (elements, cb) => {
   // Iterates over a list of elements, yielding each in turn to the `cb` function.
   // This only needs to work with arrays.
   // based off http://underscorejs.org/#each
+  for (let i = 0; i < elements.length; i++) {
+    cb(elements[i], i);
+  }
 };
 
 const map = (elements, cb) => {
   // Produces a new array of values by mapping each value in list through a transformation function (iteratee).
   // Return the new array.
+  const newArr = [];
+  for (let i = 0; i < elements.length; i++) {
+    newArr.push(cb(elements[i]));
+  }
+  return newArr;
 };
 
 const reduce = (elements, cb, memo = elements.shift()) => {
   // Combine all elements into a single value going from left to right.
   // Elements will be passed one by one into `cb`.
   // `memo` is the starting value.  If `memo` is undefined then make `elements[0]` the initial value.
+  if (memo === undefined) memo = elements[0];
+  for (let i = 0; i < elements.length; i++) {
+    memo = cb(memo, elements[i]);
+  }
+  return memo;
 };
 
 const find = (elements, cb) => {
   // Look through each value in `elements` and pass each element to `cb`.
   // If `cb` returns `true` then return that element.
   // Return `undefined` if no elements pass the truth test.
+  for (let i = 0; i < elements.length; i++) {
+    if (cb(elements[i]) === true) return elements[i];
+  }
 };
 
 const filter = (elements, cb) => {
   // Similar to `find` but you will return an array of all elements that passed the truth test
   // Return an empty array if no elements pass the truth test
+  const newArr = [];
+  for (let i = 0; i < elements.length; i++) {
+    if (cb(elements[i]) === true) {
+      newArr.push(elements[i]);
+    }
+  }
+  return newArr;
 };
 
 const flatten = (elements) => {
   // Flattens a nested array (the nesting can be to any depth).
   // Example: flatten([1, [2], [3, [[4]]]]); => [1, 2, 3, 4];
+  // iterate through the array
+  // determine if each element is an array or not
+  // concat arrays down into elements
+  // push all non-arrays into a new array
+  const newArr = [];
+  for (let i = 0; i < elements.length; i++) {
+    if (Array.isArray(elements[i])) {
```

Comment:
This implementation will only flatten nested arrays one level deep. If your code runs into a twice-nested array or more, it will unwrap the outer-most level of array nesting and then concatenate the rest, leaving you with a once-nested array. 

What needs to happen is `flatten` needs to recursively check each array element, and continue checking all levels of nesting until it finally reaches an element that is not an array, and then concatenate that element. 

Here's an example implementation that makes use of the `reduce` function that was implemented earlier:
```
const flatten = (array) => {
    return reduce(array, (a, b) => a.concat(Array.isArray(b) ? flatten(b) : b), []);
};
```
Probably not a very intuitive implementation, so let's break this down. Recall that `reduce` takes a list of elements, a callback to pass those elements to, and a memo, which is just an empty array in this case. It's the same as how you initialized a new array at the beginning of your function. So we're passing our initial array to reduce as its first argument, and then our callback checks that the next element we're looking at is an array or not. If so, we continue to recursively call this `flatten` function, so that we can strip away every level of nesting, until we finally get an element that is not an array, and then concatenate that onto our memo array.